src/project/environment.rs

```diff
@@ -321,6 +339,28 @@ mod tests {
             .tasks(Some(Platform::Osx64))
             .is_err())
     }
+    #[test]
+    fn test_filtered_tasks() {
+        let manifest = Project::from_str(
+            Path::new("pixi.toml"),
+            r#"
+        [project]
+        name = "foobar"
+        channels = []
+        platforms = ["linux-64"]
```

Comment:
```suggestion
        platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
```
For the environment creation the current platform has to be supported. This should fix that. 

Nice that you found a clean way to add a test!

Reply:
Thank you